scripts/deploy.sh

```diff
@@ -1,31 +1,58 @@
 #!/usr/bin/env bash
 
 # Exit in case of error
-set -e
+set -euo pipefail
 
-[ "${DOMAIN}" = '' ] && echo "ERROR: Remember to set DOMAIN. e.g. DOMAIN=cops-staging.openstax.org" && exit 1
-[ "${STACK_NAME}" = '' ] && echo "ERROR: Remember to set STACK_NAME. e.g. STACK_NAME=cops-stag" && exit 1
-[ "${TAG}" = '' ] && echo "ERROR: Remember to set TAG." && exit 1
+if test -n "${DEBUG_RUN_LOCAL:-}"; then
+    if test -n "${SSH_HOST:-}"; then
+        echo "ERROR: SSH_HOST and DEBUG_RUN_LOCAL should not both be set"
+        exit 111
+    fi
+    run_script() {
+        env -i HOME="$HOME" PATH="$PATH" bash -seuxo pipefail
+    }
+else
+    run_script() {
+        ssh "${SSH_HOST:?}" 'bash -seuo pipefail'
+    }
+fi
 
+: "${DOMAIN:?'Remember to set DOMAIN. e.g. DOMAIN=corgi-staging.openstax.org'}"
+: "${STACK_NAME:?'Remember to set STACK_NAME. e.g. STACK_NAME=corgi-stag'}"
+: "${TAG:?'Remember to set TAG'}"
 REVISION=$(git --git-dir=./.git rev-parse --short HEAD)
 DEPLOYED_AT=$(date '+%Y%m%d.%H%M%S')
 
-echo "DOMAIN=${DOMAIN}
-STACK_NAME=${STACK_NAME}
-TAG=${TAG}
-REVISION=${REVISION}
-DEPLOYED_AT=${DEPLOYED_AT}"
+echo "Now updating $STACK_NAME with: 
+TAG=$TAG
+REVISION=$REVISION
+DOMAIN=$DOMAIN
+DEPLOYED_AT=$DEPLOYED_AT"
 
-set +x
-DOMAIN=${DOMAIN} \
-STACK_NAME=${STACK_NAME} \
-TAG=${TAG} \
-REVISION=${REVISION} \
-DEPLOYED_AT=${DEPLOYED_AT} \
-SUBNET=${SUBNET} \
-SESSION_SECRET=${SESSION_SECRET} \
-docker-compose \
--f docker-compose.stack.release.yml \
-config > docker-stack.yml
-
-docker -H ssh://corgi stack deploy -c docker-stack.yml --with-registry-auth "${STACK_NAME}"
+base64_gzipped_compose_file="$(gzip -kc "${COMPOSE_FILE_PATH:?}" | base64)"
```

Comment:
Nice! (zip and base64-encode the whole docker-compose file, and send it as part of the SSH command)